etc/docs/api.proto

```diff
@@ -0,0 +1,62 @@
+service ExecutorService{
+  rpc ExecuteCommand (ExecuteRequest) returns (ExecuteReply) {}
+  rpc StopExecution (ExecutionIdRequest) returns (StopCommandReply) {}
+  rpc ListExecutions (EmptyRequest) returns (ExecutionListReply) {}
+  rpc GetExecution (ExecutionIdRequest) returns (ExecutionInfoReply) {}
+  rpc GetOutStream(ExecutionIdRequest) returns (stream OutChunk) {}
+}
+
+message ExecuteRequest {
+  required string command = 1;
+  repeated string args = 2;
+}
+
+message ExecutionIdRequest {
+  required ExecutionId executionId = 1;
+}
+
+message EmptyRequest {}
+
+message ExecutionId {
+  required string value = 1;
+}
+
+message ExecuteReply {
+  required ExecutionId executionId = 1;
+}
+
+message ExecutionListReply {
+  repeated ExecutionInfoReply executions = 1;
+}
+
+message ExecutionInfoReply {
+  required ExecutionId executionId = 1;
+  enum ExecutionStatus {
+    UNKNOWN = 0;
+    FINISHED = 1;
+    RUNNING = 2;
+    ABORTED = 3;
+    NOT_STARTED = 4;
```

Comment:
what do these two statuses mean?

Reply:
NOT_STARTED means that the execution failed to start, this would be for example trying to start an executable that does not exist.

ABORTED means that the process was started at some point, but since it has finished and had a non-zero exit code. For example the user SIGKILL-ing the process will result in an ABORTED state (this means we lose the distinction between SIGTERM and SIGKILL at a status level, but I believe with having the exit code that's fine)